contracts/FlashMintLib.sol

```diff
@@ -0,0 +1,154 @@
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "./Interfaces/Compound/CErc20I.sol";
+import "./Interfaces/Compound/ComptrollerI.sol";
+import "./Interfaces/DAI/IERC3156FlashLender.sol";
+import "./Interfaces/DAI/IERC3156FlashBorrower.sol";
+
+interface IWETH is IERC20 {
+	function deposit() payable external;
+	function withdraw(uint256) external;
+}
+
+interface IUniswapAnchoredView {
+	function price(string memory) external returns (uint);
+}
+
+interface IERC20Extended is IERC20 {
+	function decimals() external view returns (uint8);
+
+	function name() external view returns (string memory);
+
+	function symbol() external view returns (string memory);
+}
+
+library FlashMintLib {
+	using SafeMath for uint256;
+	event Leverage(uint256 amountRequested, uint256 amountGiven, bool deficit, address flashLoan);
+
+	uint256 private constant PRICE_DECIMALS = 1e6;
+	uint256 private constant DAI_DECIMALS = 1e18;
+	uint256 private constant COLLAT_RATIO_DAI = 0.74 ether;
+    address public constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address public constant CDAI = 0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643;
+	address private constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+	address private constant WBTC = 0x2260FAC5E5542a773Aa44fBCfeDf7C193bc2C599;
+	ComptrollerI private constant COMP = ComptrollerI(0x3d9819210A31b4961b30EF54bE2aeD79B9c9Cd3B);
+    address public constant LENDER = 0x1EB4CF3A948E7D72A198fe073cCb8C7a948cD853;
+    bytes32 public constant CALLBACK_SUCCESS = keccak256("ERC3156FlashBorrower.onFlashLoan");
+
+    function doFlashMint(bool deficit, uint256 amountDesired, address want, uint256 maxFee) public returns (uint256) {
+        if(amountDesired == 0) {
+            return 0;
+        }
+
+        // calculate amount of ETH we need
+        (uint256 requiredDAI, uint256 amountWant) = getFlashLoanParams(want, amountDesired);
+
+        bytes memory data = abi.encode(deficit, amountWant);
+        uint256 _allowance = IERC20(DAI).allowance(address(this), address(LENDER));
+        uint256 _fee = IERC3156FlashLender(LENDER).flashFee(DAI, amountWant);
+        // Check that fees have not been increased without us knowing
+        require(_fee <= _calcFee(requiredDAI, maxFee));
+        uint256 _repayment = requiredDAI.add(_fee);
+        if(_allowance < _repayment) {
+            IERC20(DAI).approve(address(LENDER), 0);
+            IERC20(DAI).approve(address(LENDER), type(uint256).max);
+        }
+        IERC3156FlashLender(LENDER).flashLoan(IERC3156FlashBorrower(address(this)), DAI, requiredDAI, data);
+
+        emit Leverage(amountDesired, requiredDAI, deficit, address(LENDER));
+
+        return amountWant;
+    }
+
+    function maxLiquidity() public view returns (uint256) {
+        return IERC3156FlashLender(LENDER).maxFlashLoan(DAI);
+    }
+
+    function _calcFee(uint256 amount, uint256 maxFee) internal view returns (uint256) {
```

Comment:
the flashloan logic does not work with any fee as you dont redeem an amount to pay the fee. you may as well just do a check that fee is zero. 

To work with fee you need to separate out amount and repayamount. See below if you're interested.

````
_loanLogic(deficit, amount, amount.add(_fee));

function _loanLogic(
        bool deficit,
        uint256 amount,
        uint256 repayAmount
    ) internal {
        uint256 bal = want.balanceOf(address(this));
        require(bal >= amount); // to stop malicious calls
        //if in deficit we repay amount and then withdraw
        if (deficit) {
            cToken.repayBorrow(amount);
            //if we are withdrawing we take more to cover fee
            cToken.redeemUnderlying(repayAmount);
        } else {
            //check if this failed incase we borrow into liquidation
            require(cToken.mint(bal) == 0, "mint error");
            //borrow more to cover fee
            // fee is so low for dydx that it does not effect our liquidation risk.
            //DONT USE FOR AAVE
            cToken.borrow(repayAmount);
        }
    }
```
`


**FEE can get us liquidated when leveraging up**